src/vm/gchelpers.cpp

```diff
@@ -1314,7 +1314,7 @@ void ErectWriteBarrierForMT(MethodTable **dst, MethodTable *ref)
 
     *dst = ref;
 
-#ifdef _DEBUG
+#ifdef WRITE_BARRIER_CHECK
```

Comment:
Thank you for the explanation.
I removed the redundant definition of `WRITE_BARRIER_CHECK` in `clr.props`.
I couldn't figure out how to add the `WRITE_BARRIER_CHECK` definition for CMake build to be under `src\vm` so I just updated src/pal/tools/gen-buildsys-clang.sh and src/pal/tools/windows-compiler-override.txt to define `WRITE_BARRIER_CHECK` the same way as for debug builds.